gcp_variant_transforms/transforms/infer_headers_test.py

```diff
@@ -401,3 +417,141 @@ def test_pipeline(self):
       assert_that(inferred_headers,
                   asserts.header_fields_equal_ignore_order([expected]))
       p.run()
+
+  def test_infer_annotation_types_no_conflicts(self):
+    anno_fields = ['CSQ']
+    header = self._get_sample_header_fields_with_annotation()
+    variant = self._get_sample_variant_1()
+    variant.info['CSQ'] = ['A|GENE1|100|1.2', 'TT|GENE1|101|1.3']
+    infer_header_fields = infer_headers._InferHeaderFields(False, anno_fields)
+    new_header = next(infer_header_fields.process(variant, header))
+    desc = 'Inferred type field for annotation {}.'
+    expected = vcf_header_io.VcfHeader(infos={
+        'CSQ_Gene_TYPE':
+        Info('CSQ_Gene_TYPE', 1, 'String', desc.format('Gene'), '', ''),
+        'CSQ_Position_TYPE':
+        Info('CSQ_Position_TYPE', 1, 'Integer',
+             desc.format('Position'), '', ''),
+        'CSQ_Score_TYPE':
+        Info('CSQ_Score_TYPE', 1, 'Float', desc.format('Score'), '', '')})
+    self.assertEqual(expected, new_header)
+    return new_header, expected
+
+  def test_infer_annotation_types_with_conflicts(self):
```

Comment:
nit: s/_conflicts/_type_conflicts (it may not be obvious that you're referring to type conflicts (it could be name or other things)).